users/users.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package users
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/absmach/magistrala/pkg/errors"
+	"github.com/absmach/magistrala/pkg/postgres"
+	"golang.org/x/net/idna"
+)
+
+const (
+	maxLocalLen  = 64
+	maxDomainLen = 255
+	maxTLDLen    = 24 // longest TLD currently in existence
+
+	atSeparator  = "@"
+	dotSeparator = "."
+)
+
+var (
+	userRegexp    = regexp.MustCompile("^[a-zA-Z0-9!#$%&'*+/=?^_`{|}~.-]+$")
+	hostRegexp    = regexp.MustCompile(`^[^\s]+\.[^\s]+$`)
+	userDotRegexp = regexp.MustCompile("(^[.]{1})|([.]{1}$)|([.]{2,})")
+)
+
+type User struct {
+	ID             string      `json:"id"`
+	FirstName      string      `json:"first_name,omitempty"`
+	LastName       string      `json:"last_name,omitempty"`
+	Tags           []string    `json:"tags,omitempty"`
+	Metadata       Metadata    `json:"metadata,omitempty"`
+	CreatedAt      time.Time   `json:"created_at,omitempty"`
+	UpdatedAt      time.Time   `json:"updated_at,omitempty"`
+	UpdatedBy      string      `json:"updated_by,omitempty"`
+	Status         Status      `json:"status,omitempty"` // 1 for enabled, 0 for disabled
+	Role           Role        `json:"role,omitempty"`   // 1 for admin, 0 for normal user
+	ProfilePicture string      `json:"profile_picture,omitempty"`
```

Comment:
Add a comment explaining this is a URL. Also consider chaning type to `url.URL`.